### PR TITLE
Fix tunnel address for TLS routing if public tunnel address is present

### PR DESCRIPTION
**Pull request by smallinsky**
Issue: https://github.com/gravitational/teleport/issues/8952

## What
Fix destination reverse tunnel address in case if proxy support TLS Routing. 

This PR change the `tunnelAddr` logic and only  `public_proxy_addr` and `webProxy` addresses  are used as tunnel address if proxy support TLS Routing.

### CODE REVIEW

File: api/client/webclient/webclient.go [hunk 1]
```diff
@@ -376,6 +374,38 @@ func tunnelAddr(proxyAddr string, settings ProxySettings) (string, error) {
...
+		return proxyAddr, nil
+	}
+	host, err := extractHost(proxyAddr)
+	if err != nil {
+		return "", trace.Wrap(err, "failed to parse the given proxy address")
+	}
+
+	// Got proxy address without a port like:  proxy.example.com
+	// If proxyAddr doesn't contain any port it means that HTTPS port should be used.
+	return net.JoinHostPort(host, "443"), nil
```
Comment (r0mant) [inline api/client/webclient/webclient.go:406]: Why 443? Isn't default proxy port 3080?

Comment (smallinsky) [reply api/client/webclient/webclient.go:406]: This code is invoked only if `Find web` API  call succeed. 


if `proxyAddr` format doesn't contain port number it means that the default `https` port was used in Find API call: 
```
	endpoint := fmt.Sprintf("https://%s/webapi/find", proxyAddr)
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, endpoint, nil)
```

Comment (r0mant) [reply api/client/webclient/webclient.go:406]: Hmm I see. This is a little confusing, let's at least expand the comment to explain. Also, do we have a constant for 443?

**Review by r0mant — APPROVED** [review]

File: api/client/webclient/webclient.go (hunk 1 above)
Comment (smallinsky) [reply api/client/webclient/webclient.go]: Extended the comment.  
>  Also, do we have a constant for 443?
there is gravitational/teleport.StandardHTTPSPort const but importing `gravitational/teleport`  in teleport/api` pkg is not possible thus I move this const to api defaults.

**Review by Joerger — APPROVED** [review]

**Pull request merged**